### DISCUSSION TITLE
Not able to load session with mini.sessions

### DISCUSSION THREAD

**Discussion by rbhanot4739** [nvim-mini/mini.nvim#1289] [Q&A] [answered]
Hello,

I installed the plugin with the following config
```
  {
    "echasnovski/mini.sessions",
    version = false,
    config = function()
      require("mini.sessions").setup({ autoread = true, verbose = { read = true, write = true, delete = true } })
    end,
  }
```
So I launch nvim and open bunch of files and then quit it. Upon launching vim the session isn't loaded automatically and I get the following message
```
14:39:08 msg_show (mini.sessions) There is no detected sessions. Change configuration and rerun `MiniSessions.setup()`.
```
This is how my sessionopts look like `sessionoptions=buffers,curdir,tabpages,winsize,help,globals,skiprtp,folds`
Am i missing something in setting up the plugin ?

**Comment by echasnovski** [comment, marked as answer]
Everything seems to be correctly set up, just a bit of workflow is missing.

There is no detected sessions because there was none written yet. You have to explicitly create (i.e. write) at least one in order for `autoread` to pick it up. In other words:
- Open Neovim with a bunch of files. This does not yet constitutes a "current session".
- Execute `:lua MiniSessions.write('my-session')` (choose your preferred session name). This session now constitutes a "current session".
- You can open another set of files to see the effect of the default `autowrite = true`.
- Close Neovim and reopen. It should now load the previous set of files including the ones opened after `MiniSessions.write()` call.

**Reply by rbhanot4739** [reply]
Thank you @echasnovski for the response, however isn't `autowrite = true` supposed to write the session, same thing mentioned `Autowrite current session before quitting Neovim.` in readme ? Or the pre-req for this to manually save a session first and then `autowrite` would keep writing the subsequent sessions ?

**Reply by echasnovski** [reply]
This was a point of confusion several times already, so I dedicated some time to clarifying this point in the help and README. They [now have the wording](https://github.com/echasnovski/mini.nvim/blob/9618cf56e76cc0caab54ad60116a8d72f99d7b3b/doc/mini-sessions.txt#L33) of "Autowrite currently read session before quitting Neovim.". I hope this should make a bit clearer that just opening files without them being part of already read (i.e. loaded) session won't autowrite them. The main reason for that is that that session should have name explicitly set by the user.
